### ISSUE TITLE
The disappearing progress bar

### ISSUE THREAD

**Issue by pickfire** [mitsuhiko/indicatif#289]
I have some issues when trying to use indicatif with rayon and I end up doing my own painting. Here is the minimum reproducible thing.

```rust
// indicatif = { version = "0.16.1", features = ["rayon"] }
// rayon = "1.5.1"
use indicatif::ParallelProgressIterator;
use rayon::prelude::*;
use std::thread;
use std::time::Duration;

fn main() {
    let v: Vec<_> = (0..20).collect();
    let n = v
        .par_iter()
        .progress()
        .map(|x| {
            thread::sleep(Duration::from_secs(1));
            x % 2 == 0
        })
        .count();
    println!("{}", n);
}                                                        
```

![output-opt](https://user-images.githubusercontent.com/4687791/119075091-cc0d2500-ba22-11eb-8671-f27bb6cc44dd.gif)

**Comment by pickfire** [comment]
Thanks it was fixed.

**Comment by djc** [comment]
Released this as part of 0.16.2.
